include/sys/cbprintf_internal.h

```diff
@@ -120,6 +120,26 @@ extern "C" {
  *
  * @param arg Argument.
  */
+#ifdef __sparc__
+static inline void cbprintf_wcpy(int *dst, int *src, uint32_t len)
+{
+	for (int i = 0; i < len; i++) {
+		dst[i] = src[i];
+	}
+}
```

Comment:
Yes, all `<stdarg.h>` variable list arguments are 32-bit aligned on the stack. So it should be fine.

The SPARC V8 ABI specification, page 3-36, describes how variable list arguments are stored, together with an example.
https://www.gaisler.com/doc/sparc-abi.pdf#page=50


I would recommend that the first two sentences of the commit message are rephrased, in line with:

```
The SPARC V8 ABI guarantees that the arguments of a variable argument
list function are stored on the stack at addresses which are 32-bit
aligned. It means that variables of type unit64_t and double may not
be properly aligned on the stack.

The compiler is aware of the ABI and takes care of this. However,
as we are directly accessing the variable argument list here, we need
to take the alignment into consideration and copy 64-bit arguments
as 32-bit words.
```

Reply:
I've added this as a comment in the code when defining a special treatment in #33473